src/CLI/OutputGenerator.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Axe.Windows.Automation;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace AxeWindowsScanner
+{
+    public class OutputGenerator : IOutputGenerator
+    {
+        private readonly TextWriter _writer;
+
+        private bool _bannerHasBeenShown;
+
+        public OutputGenerator(TextWriter writer)
+        {
+            if (writer == null) throw new ArgumentNullException(nameof(writer));
+
+            _writer = writer;
+        }
+
+        public void ShowOutput(IOptions options, IErrorCollector errorCollector, ScanResults scanResults)
+        {
+            bool failedToComplete = errorCollector.ParameterErrors.Any() || errorCollector.Exceptions.Any() || (scanResults == null);
+
+            ShowBanner(options, failedToComplete ? VerbosityLevel.Quiet : VerbosityLevel.Default);
```

Comment:
Why is the banner dependent on the verbosity? Isn't this controlled by the options?

Reply:
Come to think of it, why is the banner shown here? Why is the banner not shown before anything else has run?

Reply:
The banner is omitted in quiet mode if execution completes normally. In non-quiet modes, we call ShowBanner immediately (to provide feedback that the app is doing something). We call ShowBanner again if we failed to complete, but we can't do that until after the app has run. This is to balance both scenarios.

Reply:
"We call ShowBanner again..." How come?

Reply:
Discussed offline